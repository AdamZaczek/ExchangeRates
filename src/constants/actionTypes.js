export const FETCH_CURRENCIES = 'FETCH_CURRENCIES';
export const ADD_CURRENCY = 'ADD_CURRENCY';
export const REMOVE_CURRENCY = 'FETCH_CURRENCY';
export const REMOVE_CURRENCIES = 'REMOVE_CURRENCIES';
