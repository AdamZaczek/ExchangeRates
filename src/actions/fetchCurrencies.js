import { FETCH_CURRENCIES } from '../constants/actionTypes';

const fetcCurrenciesSuccess = currencies => {
  return {
    type: FETCH_CURRENCIES,
    currencies: currencies[0].rates,
  }
};

export const fetchCurrencies = (url) => async (dispatch) => {
    const rateRes = await fetch(url);
    const resToJson = await rateRes.json();
    dispatch(fetcCurrenciesSuccess(resToJson));
};
