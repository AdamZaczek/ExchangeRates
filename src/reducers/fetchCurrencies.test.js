import fetchCurrencies from './fetchCurrencies';
import { FETCH_CURRENCIES } from '../constants/actionTypes';

describe('fetchCurrencies reducer', () => {
  test('should return default state', () => {
    expect(fetchCurrencies(undefined, {})).toMatchSnapshot();
  });

  test('should return an array with currencies', () => {
    const c1 = {
      code: '1',
      currency: 'dollar'
    }
    const c2 = {
      code: '2',
      currency: 'euro'
    }
    const c3 = {
      code: '3',
      currency: 'frank'
    }
    const action = {
      type: FETCH_CURRENCIES,
      currencies: [c1, c2, c3]
    };
    expect(fetchCurrencies({ currencies: [] }, action)).toMatchSnapshot();
	});
});