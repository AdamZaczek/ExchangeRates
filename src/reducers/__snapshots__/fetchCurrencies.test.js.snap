// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchCurrencies reducer should return an array with currencies 1`] = `
Object {
  "currencies": Array [
    Object {
      "code": "1",
      "currency": "dollar",
    },
    Object {
      "code": "2",
      "currency": "euro",
    },
    Object {
      "code": "3",
      "currency": "frank",
    },
  ],
}
`;

exports[`fetchCurrencies reducer should return default state 1`] = `
Object {
  "currencies": Array [],
}
`;
